### DISCUSSION TITLE
Catching window move event

### DISCUSSION THREAD

**Discussion by WhoAteDaCake** [lc-soft/LCUI#246] [Q&A] [answered]
Hello, is there a way to detect when a window was moved ? And how to move it programatically.
I noticed that `Surface_Move` function can be used to move the window, but I couldn't find any code besides `LCUIDisplay_BindSurface` calling it directly. Binding to `"surface"` event doesn't get triggered on move either.

Thanks for the help!

**Comment by lc-soft** [comment, marked as answer]
> is there a way to detect when a window was moved ? 

In Linux, you can call `LCUI_BindSysEvent()` function to listen for the `ConfigureNotify` event, which contains the location and size of the window.

Related documents:

- [Chapter 10: Events
](https://tronche.com/gui/x/xlib/events/)
- [10.10.2 ConfigureNotify](https://tronche.com/gui/x/xlib/events/window-state-change/configure.html)

> And how to move it programatically.

In the original design, the position and size of the widget will be synchronized to the surface it is bound to, but there is a problem that the window cannot move smoothly, so I cancel the position synchronization.

Can you tell me why you need to capture window move event?

**Reply by WhoAteDaCake** [reply]
It's for convenience sake. I'm experiment to see if it's useful to open the window back from the same position it was closed in.

So for example if I had my email client on the right side of the window taking up half of the screen, when I close it, then open it again, it should be in the right position, taking up half of the screen

**Reply by WhoAteDaCake** [reply]
I guess rather than catching the move event, I should catch the `close` event and save the position from there

**Reply by lc-soft** [reply]
> I guess rather than catching the move event, I should catch the `close` event and save the position from there

OK, I will take this into consideration when I developing the surface module of LCUI version 3.0.
